cmd/clusterctl/examples/aws/cluster-network-spec.yaml.template

```diff
@@ -0,0 +1,20 @@
+apiVersion: "cluster.k8s.io/v1alpha1"
+kind: Cluster
+metadata:
+  name: ${CLUSTER_NAME}
+spec:
+  clusterNetwork:
+    services:
+      cidrBlocks: ["10.96.0.0/12"]
+    pods:
+      cidrBlocks: ["192.168.0.0/16"]
+    serviceDomain: "cluster.local"
+  providerSpec:
+    value:
+      apiVersion: "awsprovider/v1alpha1"
+      kind: "AWSClusterProviderSpec"
+      region: "${AWS_REGION}"
+      sshKeyName: "${SSH_KEY_NAME}"
+      networkSpec:
+        vpc:
+          id: "${VPC_ID}"
```

Comment:
Instead of having a completely separate (and mostly duplicated) template here, would it make sense to use kustomize for adding the networkSpec to the existing template for this purpose instead?

Reply:
I thought about it, but Kustomize isn't powerful enough to allow for user-input or get environment flags. I've seen other projects having the same issue and using `sed` to overcome these problems. I think in the future we should template kustomize files instead using maybe something like go templates, or similar environment substitution tool.

Reply:
The issue with trying to override env vars at runtime is a known issue with kustomize (and one that they don't currently plan on supporting), we work around a similar issue to this in how we manage the aws_manager_image_patch.yaml file.

Reply:
Opened https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/549 to track :)